File: quests/nurga/a_sleeping_ogre.pl
# Monk Epic 1.0
sub EVENT_SPAWN {
  quest::settimer("appearance", 1);
}

sub EVENT_TIMER {
  quest::stoptimer("appearance");
  $npc->SetAppearance(3);
}

sub EVENT_SAY {
  if($text=~/hail/i) {
    quest::emote("yawns, and utters some gibberish, then falls back asleep.");
  }
}

sub EVENT_ITEM {
  if((plugin::check_handin(\%itemcount, 1685 => 1)) && ($ulevel >= 46) && ($class eq "Monk")) { #Breath of Gwan
    $npc->SetAppearance(0);
    quest::say("Your path of wanton destruction ends here. Gwan and Eejag were impatient and hot-headed. You will not defeat me, for I have the patience and perseverance of stone, unlike the children you have beaten before me. Are you sure you want to challenge me?");
    $trunt = quest::spawn2(107161,0,0,$x,$y,$z,$h); #Trunt
    $npcobj = $entity_list->GetMobID($trunt);
    $truntnpc = $npcobj->CastToNPC();
    $truntnpc->AddToHateList($client,1);
    quest::depop();
  }
  else {
    plugin::return_items(\%itemcount);
  }
}
# End of File

File: nurga/a_sleeping_ogre.lua
-- Monk Epic 1.0
function event_spawn(e)
	eq.set_timer("appearance", 1000);
end

function event_timer(e)
	eq.stop_timer("appearance");
	e.self:SetAppearance(3);
end

function event_say(e)
	if(e.message:findi("hail")) then
		e.self:Emote(eq.ChooseRandom("snorts a bit and continues sleeping.","yawns, and utters some gibberish, then falls back asleep."))
	end
end

function event_trade(e)
	local item_lib = require("items");
	if(item_lib.check_turn_in(e.trade, {item1 = 1685})) then --Breath of Gwan
		e.self:SetAppearance(0);
		e.self:Say("Your path of wanton destruction ends here. Gwan and Eejag were impatient and hot-headed. You will not defeat me, for I have the patience and perseverance of stone, unlike the children you have beaten before me. Are you sure you want to challenge me?");
		eq.spawn2(107161,0,0,e.self:GetX(),e.self:GetY(),e.self:GetZ(),e.self:GetHeading()):AddToHateList(e.other,1); --Trunt
		eq.depop_with_timer();
	end
	item_lib.return_items(e.self, e.other, e.trade)
end
